src/collections/vector.h

```diff
@@ -35,13 +44,40 @@ template <typename T>
 inline Vec<T>::Vec() {
   this->buf = nullptr;
   len = 0;
+  _capacity = 0;
+}
+
+template <typename T>
+void Vec<T>::reallocate() {
```

Comment:
this function should be more reusable, try to add a parameter called `resize` and just resize the underlying container(maybe change the `this->_capacity` with side effects.)